incubator/group/proposal_executor_test.go

```diff
@@ -0,0 +1,108 @@
+package group
+
+import (
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/baseapp"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/stretchr/testify/require"
+)
+
+func TestDoExecuteMsgs(t *testing.T) {
+	specs := map[string]struct {
+		srcAccount sdk.AccAddress
+		srcMsgs    []sdk.Msg
+		srcHandler sdk.Handler
+		expErr     bool
+	}{
+		"all good": {
+			srcAccount: []byte("my-group-acct-addrss"),
+			srcMsgs:    []sdk.Msg{MyMsg{[]sdk.AccAddress{[]byte("my-group-acct-addrss")}}},
+			srcHandler: mockHandler(&sdk.Result{}, nil),
+		},
+		"not authz by group account": {
+			srcAccount: []byte("my-group-acct-addrss"),
+			srcMsgs:    []sdk.Msg{MyMsg{[]sdk.AccAddress{[]byte("any--other---address")}}},
+			srcHandler: alwaysPanicHandler(),
```

Comment:
shouldn't this be `mockHandler` if we actually want to test this case? (panic might get caught by the router)

Reply:
In this scenario the authentication decorator is tested. The handler should not be called as the payload message "signer" is not the groupAccount. This prevents access to random accounts.
The `alwaysPanicHandler` makes sense as it would raise a panic and not an error. The panic is not handled by the router.